pkg/client/build.go

```diff
@@ -1037,14 +1037,13 @@ func (c *Client) fetchBuildpack(ctx context.Context, bp string, relativeBaseDir
 			return nil, nil, errors.Wrapf(err, "getting builder OS")
 		}
 
-		builderArch, err := builderImage.Architecture()
 		if err != nil {
 			return nil, nil, errors.Wrapf(err, "getting builder architecture")
 		}
 		downloadOptions := buildpack.DownloadOptions{
 			RegistryName:    registry,
 			ImageOS:         builderOS,
-			Platform:        fmt.Sprintf("%s/%s", builderOS, builderArch),
+			Platform:        builderOS,
```

Comment:
The `was found but does not match the specified platform` error is coming from the daemon (see [here](https://github.com/moby/moby/blob/5bff90cea2a77446163019e00016f5e952f591c1/daemon/images/image.go#L234)). Prior to #1933 we never set any platform (see [here](https://github.com/buildpacks/pack/pull/1933/files#diff-a0c732d7543d7450a4886f604425cdd9195ca3da77338f9a0f4b6980a6ad5eb6L328)) so I think the platform provided [here](https://github.com/buildpacks/pack/blob/b12c9b3a0a8b3e9e0df590eaf901af44e5c82f06/pkg/image/fetcher.go#L181) was an empty string. Rather than override the expected platform, I wonder if we could inspect the error and retry mismatched platform with empty string.

Reply:
I changed the implementation to retry on failure